File: Projeto_6/professor.cpp
#include "Professor.h"
#include <iostream>

Professor::Professor(const std::string& nome, const std::string& email, const std::string& areaDeAtuacao, const std::string& senha)
    : UsuarioAutenticavel(nome, email, "PROFESSOR", senha), areaDeAtuacao(areaDeAtuacao) {}

void Professor::adicionarDisciplina(const std::string& disciplina) {
    disciplinasMinistradas.push_back(disciplina);
}

void Professor::gerarRelatorio() const {
    std::cout << "Relatório do Professor\n";
    std::cout << "Nome: " << nome << "\nEmail: " << email << "\nÁrea de Atuação: " << areaDeAtuacao << "\n";
    std::cout << "Disciplinas Ministradas: ";
    for (const auto& disc : disciplinasMinistradas) {
        std::cout << disc << " ";
    }
    std::cout << "\n";
}

bool Professor::autenticar(const std::string& senhaTentativa) {
    return senhaTentativa == senha;
}

File: Projeto_6/src/professor.py
from usuario_autenticavel import UsuarioAutenticavel
from relatorio import Relatorio
from tipo_usuario import TipoUsuario

class Professor(UsuarioAutenticavel, Relatorio):
    def __init__(self, nome, login, senha, disciplina):
        super().__init__(nome, login, senha, TipoUsuario.PROFESSOR)
        self.disciplina = disciplina

    def autenticar(self, senha: str) -> bool:
        return self._verificar_senha(senha)

    def gerar_relatorio(self):
        print(f"\nRelatório do Professor: {self.nome}")
        print(f"Disciplina: {self.disciplina}")
